### ISSUE TITLE
Test my code is not working, Showing Error: TimeoutError Process didn't seem to finish or was taking a really long time.

### ISSUE THREAD

**Issue by ManjinderSingh131** [rage/tmc-vscode#364]
I am having errors while running test. It used to work previously on pop os, In ubuntu 18.04 and windows 10 I am not able to make it work. 

My vscode details:
`
Version: 1.46.0
Commit: a5d1cc28bb5da32ec67e86cc50f84c67cc690321
Date: 2020-06-10T08:59:04.923Z
Electron: 7.3.1
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Linux x64 4.15.0-106-generic
`
 
I have also installed maven from apt-get in ubuntu. My vscode has java extension pack installed.

I am having these errors, while I try to run tests. 
`TimeoutError Process didn't seem to finish or was taking a really long time.`
Can you provide any hint to solve this issue please? I was loving this extension and had finished 1 part.

I had tried purging local maven that this extension installs. 

`[2020-06-18 11:51:07:242] ERROR: Exercise test run failed: TimeoutError - Process didn't seem to finish or was taking a really long time. 
[2020-06-18 11:51:07:451] ERROR: Process raised error: java -jar "/home/manjinder/.config/Code/User/globalStorage/moocfi.test-my-code/tmcdata/tmc-langs-cli-0.7.17-SNAPSHOT.jar" run-tests --exercisePath="/home/manjinder/.config/Code/User/globalStorage/moocfi.test-my-code/tmcdata/TMC workspace/Exercises/mooc/mooc-java-programming-i/part02-Part02_01.Squared" --outputPath="/home/manjinder/.config/Code/User/globalStorage/moocfi.test-my-code/tmcdata/temp_1.json" — {"killed":false,"code":null,"signal":"SIGTERM","cmd":"java -jar \"/home/manjinder/.config/Code/User/globalStorage/moocfi.test-my-code/tmcdata/tmc-langs-cli-0.7.17-SNAPSHOT.jar\" run-tests --exercisePath=\"/home/manjinder/.config/Code/User/globalStorage/moocfi.test-my-code/tmcdata/TMC workspace/Exercises/mooc/mooc-java-programming-i/part02-Part02_01.Squared\" --outputPath=\"/home/manjinder/.config/Code/User/globalStorage/moocfi.test-my-code/tmcdata/temp_1.json\""}, {outputPath=/home/manjinder/.config/Code/User/globalStorage/moocfi.test-my-code/tmcdata/temp_1.json, exercisePath=/home/manjinder/.config/Code/User/globalStorage/moocfi.test-my-code/tmcdata/TMC workspace/Exercises/mooc/mooc-java-programming-i/part02-Part02_01.Squared}
2020-06-18 17:20:52 [main] INFO  Main:300 - Preparing to run tests.
2020-06-18 17:20:52 [main] INFO  ProjectType:54 - Finding plugin for /home/manjinder/.config/Code/User/globalStorage/moocfi.test-my-code/tmcdata/TMC workspace/Exercises/mooc/mooc-java-programming-i/part02-Part02_01.Squared
2020-06-18 17:20:53 [main] INFO  ProjectType:58 - Detected project as apache-maven
2020-06-18 17:20:53 [main] INFO  MavenPlugin:75 - Building maven project at /home/manjinder/.config/Code/User/globalStorage/moocfi.test-my-code/tmcdata/TMC workspace/Exercises/mooc/mooc-java-programming-i/part02-Part02_01.Squared
2020-06-18 17:20:53 [main] INFO  MavenExecutors:15 - Defaulting to MavenInvokator
2020-06-18 17:20:53 [main] INFO  MavenInvokatorMavenTaskRunner:152 - Maven bundle not previously extracted, extracting...
2020-06-18 17:20:53 [main] INFO  MavenInvokatorMavenTaskRunner:85 - Using maven at: /home/manjinder/.config/tmc/apache-maven-3.5.4
2020-06-18 17:20:55 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] Scanning for projects...
2020-06-18 17:20:55 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] 
2020-06-18 17:20:55 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] -----------------------< tkt:Part02_01.Squared >------------------------
2020-06-18 17:20:55 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] Building Part02_01.Squared 1.0-SNAPSHOT
2020-06-18 17:20:55 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] --------------------------------[ jar ]---------------------------------
2020-06-18 17:20:56 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] 
2020-06-18 17:20:56 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ Part02_01.Squared ---
2020-06-18 17:20:56 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] Deleting /home/manjinder/.config/Code/User/globalStorage/moocfi.test-my-code/tmcdata/TMC workspace/Exercises/mooc/mooc-java-programming-i/part02-Part02_01.Squared/target
2020-06-18 17:20:56 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] 
2020-06-18 17:20:56 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ Part02_01.Squared ---
2020-06-18 17:20:56 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] Using 'UTF-8' encoding to copy filtered resources.
2020-06-18 17:20:56 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] skip non existing resourceDirectory /home/manjinder/.config/Code/User/globalStorage/moocfi.test-my-code/tmcdata/TMC workspace/Exercises/mooc/mooc-java-programming-i/part02-Part02_01.Squared/src/main/resources
2020-06-18 17:20:56 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] 
2020-06-18 17:20:56 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] --- maven-compiler-plugin:3.8.0:compile (default-compile) @ Part02_01.Squared ---
2020-06-18 17:20:57 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] Changes detected - recompiling the module!
2020-06-18 17:20:57 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] Compiling 1 source file to /home/manjinder/.config/Code/User/globalStorage/moocfi.test-my-code/tmcdata/TMC workspace/Exercises/mooc/mooc-java-programming-i/part02-Part02_01.Squared/target/classes
2020-06-18 17:20:57 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] 
2020-06-18 17:20:57 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ Part02_01.Squared ---
2020-06-18 17:20:57 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] Using 'UTF-8' encoding to copy filtered resources.
2020-06-18 17:20:57 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] skip non existing resourceDirectory /home/manjinder/.config/Code/User/globalStorage/moocfi.test-my-code/tmcdata/TMC workspace/Exercises/mooc/mooc-java-programming-i/part02-Part02_01.Squared/src/main/resources
2020-06-18 17:20:57 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] 
2020-06-18 17:20:57 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] --- maven-compiler-plugin:3.8.0:compile (default-compile) @ Part02_01.Squared ---
2020-06-18 17:20:57 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] Nothing to compile - all classes are up to date
2020-06-18 17:20:57 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] 
2020-06-18 17:20:57 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ Part02_01.Squared ---
2020-06-18 17:20:57 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] Using 'UTF-8' encoding to copy filtered resources.
2020-06-18 17:20:57 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] skip non existing resourceDirectory /home/manjinder/.config/Code/User/globalStorage/moocfi.test-my-code/tmcdata/TMC workspace/Exercises/mooc/mooc-java-programming-i/part02-Part02_01.Squared/src/test/resources
2020-06-18 17:20:57 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] 
2020-06-18 17:20:57 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] --- maven-compiler-plugin:3.8.0:testCompile (default-testCompile) @ Part02_01.Squared ---
2020-06-18 17:20:57 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] Changes detected - recompiling the module!
2020-06-18 17:20:57 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] Compiling 1 source file to /home/manjinder/.config/Code/User/globalStorage/moocfi.test-my-code/tmcdata/TMC workspace/Exercises/mooc/mooc-java-programming-i/part02-Part02_01.Squared/target/test-classes
2020-06-18 17:20:58 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] ------------------------------------------------------------------------
2020-06-18 17:20:58 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] BUILD SUCCESS
2020-06-18 17:20:58 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] ------------------------------------------------------------------------
2020-06-18 17:20:58 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] Total time: 2.595 s
2020-06-18 17:20:58 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] Finished at: 2020-06-18T17:20:58+05:30
2020-06-18 17:20:58 [Thread-1] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] ------------------------------------------------------------------------
2020-06-18 17:20:58 [main] INFO  MavenPlugin:81 - Built maven project at /home/manjinder/.config/Code/User/globalStorage/moocfi.test-my-code/tmcdata/TMC workspace/Exercises/mooc/mooc-java-programming-i/part02-Part02_01.Squared
2020-06-18 17:20:58 [main] INFO  MavenPlugin:96 - Running tests for maven project at /home/manjinder/.config/Code/User/globalStorage/moocfi.test-my-code/tmcdata/TMC workspace/Exercises/mooc/mooc-java-programming-i/part02-Part02_01.Squared
2020-06-18 17:20:58 [main] INFO  MavenExecutors:15 - Defaulting to MavenInvokator
2020-06-18 17:20:58 [main] INFO  MavenInvokatorMavenTaskRunner:147 - Maven already extracted
2020-06-18 17:20:58 [main] INFO  MavenInvokatorMavenTaskRunner:85 - Using maven at: /home/manjinder/.config/tmc/apache-maven-3.5.4
2020-06-18 17:21:00 [Thread-3] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] Scanning for projects...
2020-06-18 17:21:00 [Thread-3] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] 
2020-06-18 17:21:00 [Thread-3] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] -----------------------< tkt:Part02_01.Squared >------------------------
2020-06-18 17:21:00 [Thread-3] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] Building Part02_01.Squared 1.0-SNAPSHOT
2020-06-18 17:21:00 [Thread-3] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] --------------------------------[ jar ]---------------------------------
2020-06-18 17:21:00 [Thread-3] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] 
2020-06-18 17:21:00 [Thread-3] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] >>> tmc-maven-plugin:1.12:test (default-cli) > process-test-classes @ Part02_01.Squared >>>
2020-06-18 17:21:00 [Thread-3] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] 
2020-06-18 17:21:00 [Thread-3] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ Part02_01.Squared ---
2020-06-18 17:21:00 [Thread-3] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] Using 'UTF-8' encoding to copy filtered resources.
2020-06-18 17:21:00 [Thread-3] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] skip non existing resourceDirectory /home/manjinder/.config/Code/User/globalStorage/moocfi.test-my-code/tmcdata/TMC workspace/Exercises/mooc/mooc-java-programming-i/part02-Part02_01.Squared/src/main/resources
2020-06-18 17:21:00 [Thread-3] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] 
2020-06-18 17:21:00 [Thread-3] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] --- maven-compiler-plugin:3.8.0:compile (default-compile) @ Part02_01.Squared ---
2020-06-18 17:21:01 [Thread-3] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] Nothing to compile - all classes are up to date
2020-06-18 17:21:01 [Thread-3] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] 
2020-06-18 17:21:01 [Thread-3] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ Part02_01.Squared ---
2020-06-18 17:21:01 [Thread-3] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] Using 'UTF-8' encoding to copy filtered resources.
2020-06-18 17:21:01 [Thread-3] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] skip non existing resourceDirectory /home/manjinder/.config/Code/User/globalStorage/moocfi.test-my-code/tmcdata/TMC workspace/Exercises/mooc/mooc-java-programming-i/part02-Part02_01.Squared/src/test/resources
2020-06-18 17:21:01 [Thread-3] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] 
2020-06-18 17:21:01 [Thread-3] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] --- maven-compiler-plugin:3.8.0:testCompile (default-testCompile) @ Part02_01.Squared ---
2020-06-18 17:21:01 [Thread-3] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] Nothing to compile - all classes are up to date
2020-06-18 17:21:01 [Thread-3] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] 
2020-06-18 17:21:01 [Thread-3] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] <<< tmc-maven-plugin:1.12:test (default-cli) < process-test-classes @ Part02_01.Squared <<<
2020-06-18 17:21:01 [Thread-3] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] 
2020-06-18 17:21:01 [Thread-3] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] 
2020-06-18 17:21:01 [Thread-3] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] --- tmc-maven-plugin:1.12:test (default-cli) @ Part02_01.Squared ---
2020-06-18 17:21:01 [Thread-3] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-classworlds/2.4/plexus-classworlds-2.4.pom
2020-06-18 17:21:03 [Thread-3] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-classworlds/2.4/plexus-classworlds-2.4.pom (3.9 kB at 1.9 kB/s)
2020-06-18 17:21:03 [Thread-3] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] Downloading from tmc: https://maven.mooc.fi/releases/org/codehaus/plexus/plexus-utils/2.1/plexus-utils-2.1.pom
2020-06-18 17:21:04 [Thread-3] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-utils/2.1/plexus-utils-2.1.pom
2020-06-18 17:21:05 [Thread-3] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-utils/2.1/plexus-utils-2.1.pom (4.0 kB at 7.2 kB/s)
2020-06-18 17:21:05 [Thread-3] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] Downloading from tmc: https://maven.mooc.fi/releases/org/sonatype/sisu/sisu-inject-bean/2.3.0/sisu-inject-bean-2.3.0.pom
2020-06-18 17:21:06 [Thread-3] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] Downloading from central: https://repo.maven.apache.org/maven2/org/sonatype/sisu/sisu-inject-bean/2.3.0/sisu-inject-bean-2.3.0.pom
2020-06-18 17:21:07 [Thread-3] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/sonatype/sisu/sisu-inject-bean/2.3.0/sisu-inject-bean-2.3.0.pom (7.1 kB at 15 kB/s)
2020-06-18 17:21:07 [Thread-3] INFO  MavenInvokatorMavenTaskRunner:99 - MavenInvokator: [INFO] Downloading from tmc: https://maven.mooc.fi/releases/org/sonatype/sisu/sisu-guice/3.1.0/sisu-guice-3.1.0.pom
, WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.rauschig.jarchivelib.FileModeMapper$FileSystemPreferencesReflectionChmodCommand (file:/home/manjinder/.config/Code/User/globalStorage/moocfi.test-my-code/tmcdata/tmc-langs-cli-0.7.17-SNAPSHOT.jar) to method java.util.prefs.FileSystemPreferences.chmod(java.lang.String,int)
WARNING: Please consider reporting this to the maintainers of org.rauschig.jarchivelib.FileModeMapper$FileSystemPreferencesReflectionChmodCommand
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release`

Can you help me resolving this issue? I see plugin automatically installs a local maven.

**Comment by sebazai** [comment]
Hello,

We just released a new version for TMC VSCode (v0.6.3) that bumps the timeout for running local tests from 15 seconds to one minute. 
Could you please try to update the extension to the latest version in VSCode (through the sidebar, with 4 small boxes) and try running the tests again.
If it doesn't work with the new version, are you able to provide the source code that you are using for the exercise, so we could try to reproduce the issue.

**Comment by ManjinderSingh131** [comment]
Hi, Thank you so much sebazai. This issue is fixed on both windows 10 and ubuntu 18.04.
 
I purged the old directories for TMC and then reinstalled the plugin. I think the plugin update has fixed the issue now.  Also in windows 10, I removed an environment variable for MAVEN_HOME for my other maven install that was in program files, that was also maybe causing problem that I had set previously.
 
Thank you so very much for your fixing this issue. I really appreciate that :) I am so happy now. 

Great work on the plugin.
